.github/workflows/integration-comment.yml

```diff
@@ -66,84 +66,298 @@ jobs:
-          - test_name: 'pulsar'
-            if: ${{ contains(github.event.comment.body, '/ci-run-integration-pulsar')        || contains(github.event.comment.body, '/ci-run-all') }}
-          - test_name: 'redis'
-            if: ${{ contains(github.event.comment.body, '/ci-run-integration-redis')         || contains(github.event.comment.body, '/ci-run-all') }}
-          - test_name: 'shutdown'
-            if: ${{ contains(github.event.comment.body, '/ci-run-integration-shutdown')      || contains(github.event.comment.body, '/ci-run-all') }}
-          - test_name: 'splunk'
-            if: ${{ contains(github.event.comment.body, '/ci-run-integration-splunk')        || contains(github.event.comment.body, '/ci-run-all') }}
-          - test_name: 'webhdfs'
-            if: ${{ contains(github.event.comment.body, '/ci-run-integration-webhdfs')       || contains(github.event.comment.body, '/ci-run-all') }}
+    runs-on: [linux, ubuntu-20.04-8core]
+    steps:
+      - uses: actions/checkout@v3
+
+      - run: sudo npm -g install @datadog/datadog-ci
+
+      - run: docker image prune -af ; docker container prune -f
+
+      - name: amqp
+        if: ${{ contains(github.event.comment.body, '/ci-run-integration-amqp') || contains(github.event.comment.body, '/ci-run-all') }}
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 25
+          max_attempts: 3
+          retry_on: error
+          command: cargo vdev -v int start amqp ; sleep 10 ; cargo vdev -v int test -a amqp ; cargo vdev -v int stop amqp
+
+      - name: appsignal
+        if: ${{ contains(github.event.comment.body, '/ci-run-integration-appsignal') || contains(github.event.comment.body, '/ci-run-all') }}
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 25
+          max_attempts: 3
+          retry_on: error
+          command: cargo vdev -v int start appsignal ; sleep 10 ; cargo vdev -v int test -a appsignal ; cargo vdev -v int stop appsignal
+
+      - name: aws
+        if: ${{ contains(github.event.comment.body, '/ci-run-integration-aws') || contains(github.event.comment.body, '/ci-run-all') }}
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 5
+          max_attempts: 3
+          retry_on: error
+          command: cargo vdev -v int start aws ; sleep 10 ; cargo vdev -v int test -a aws ; RET=$? ; cargo vdev -v int stop aws ; exit $RET
+
+      - name: axiom
+        if: ${{ contains(github.event.comment.body, '/ci-run-integration-axiom') || contains(github.event.comment.body, '/ci-run-all') }}
+        uses: nick-fields/retry@v2
+        with:
+          timeout_minutes: 5
+          max_attempts: 3
+          retry_on: error
+          command: cargo vdev -v int start axiom ; sleep 10 ; cargo vdev -v int test -a axiom ; RET=$? ; cargo vdev -v int stop axiom ; exit $RET
```

Comment:
This change I have moves us away from the shared workflow approach (we were doing that with `integration-test.yml`, in order to leverage the same shell context and get "free" caching. By calling into a shared workflow we lose that and thus lose the billable time savings from this change.

The alternatives I looked at were calling a local github action that we define in .github/actions. The problem with that though is that we don't have access to the secrets which are needed for some of the int tests.

However, as we speak I'm looking to see if I can export a shell function into an output from a job step, and call that in subsequent job steps. If I get that to work, the syntax would be similar to the "uses/with" , it would just be evoking that in a shell and passing the integration name.

Reply:
Ah gotcha, thanks for explaining 🙂

Reply:
👍 good questions- In hindsight this PR definitely was worthy of a bit more discourse in the opening comment about alternatives considered etc.